src/cert_store.rs

```diff
@@ -63,11 +72,145 @@ impl CertStore {
             }
         }
     }
+
+    /// Opens up the specified key store within the context of the current user.
+    ///
+    /// Known valid values for `which` are "Root" and "My".
+    pub fn open_current_user(which: &str) -> io::Result<CertStore> {
+        unsafe {
+            let data = OsStr::new(which)
+                             .encode_wide()
+                             .chain(Some(0))
+                             .collect::<Vec<_>>();
+            let store = crypt32::CertOpenStore(winapi::CERT_STORE_PROV_SYSTEM_W as winapi::LPCSTR,
+                                               0,
+                                               0,
+                                               winapi::CERT_SYSTEM_STORE_CURRENT_USER,
+                                               data.as_ptr() as *mut _);
+            if store.is_null() {
+                Err(io::Error::last_os_error())
+            } else {
+                Ok(CertStore(store))
+            }
+        }
+    }
+
+    /// Opens up the specified key store within the context of the local
+    /// machine.
+    ///
+    /// Known valid values for `which` are "Root" and "My".
+    pub fn open_local_machine(which: &str) -> io::Result<CertStore> {
+        unsafe {
+            let data = OsStr::new(which)
+                             .encode_wide()
+                             .chain(Some(0))
+                             .collect::<Vec<_>>();
+            let store = crypt32::CertOpenStore(winapi::CERT_STORE_PROV_SYSTEM_W as winapi::LPCSTR,
+                                               0,
+                                               0,
+                                               winapi::CERT_SYSTEM_STORE_LOCAL_MACHINE,
+                                               data.as_ptr() as *mut _);
+            if store.is_null() {
+                Err(io::Error::last_os_error())
+            } else {
+                Ok(CertStore(store))
+            }
+        }
+    }
+
+    /// Imports a PKCS#12-encoded key/certificate pair, returned as a
+    /// `CertStore` instance.
+    ///
+    /// The password must also be provided to decrypt the encoded data.
+    pub fn import_pkcs12(data: &[u8], password: &str) -> io::Result<CertStore> {
```

Comment:
I wouldn't know that i's possible with the windows crypto-api 
(atleast it doesn't seem to be used very often, PowerShell requires a password always).
It's possible with openssl though.